### ISSUE TITLE
Encountered an error reading input data!

### ISSUE THREAD

**Issue by Fusted** [open-keychain/open-keychain#2812]
Decrypt error.
I have imported my secret key (successful, app shows secret key), but when I try to decrypt app sends this error (from logs):
```
START] Processing input data
 [DEBUG] Attempting to process OpenPGP data
  [ERROR] Encountered an error reading input data!
```
This error repeats with importing file and coping from clipboard.

When I encrypt in app, I can read file on PC and Phone, but not the other way around

## Steps to Reproduce 
1. Import secret key (from gpg util)
2. Try to decrypt something
3. Get error

## Your Environment
* Android Version: 11
* Device Model: Realme 6 pro 
* OpenKeychain Version: 5.7.4, 5.7.5, 5.8.0
* From Google Play or F-Droid?: F-droid

**Comment by rafaelrandomize** [comment]
Hello, what openpgp software are you using on the pc to encrypt the files?

**Comment by Fusted** [comment]
> Hello, what openpgp software are you using on the pc to encrypt the files?

Hello
**gpg (GnuPG) 2.3.8**

**Comment by rafaelrandomize** [comment]
Do you get this same error on Google Play Openkeychain version 5.8.2?

**Comment by Fusted** [comment]
> Do you get this same error on Google Play Openkeychain version 5.8.2?

Yes, I ve tried on FDroid and Google play versions

**Comment by rafaelrandomize** [comment]
Well it's hard to say the cause of this error or a possible solution, I use Kleopatra on Windows 10 and Linux Mint 21.1 with GnuPG 2.4.0 for encryption and I had no problem with decryption on Openkeychain 5.8.2.

**Comment by Fusted** [comment]
> Openkeychain

Lol, it can decrypt my files from linux, but cant from macos

**Comment by rafaelrandomize** [comment]
Muito estranho, talvez valha a pena tentar com o GPG Suite 2022.2 for Mac OS by gpgtools.org, ele usa uma versão mais antiga do gpg, a 2.2.40, mas se a versão 2.3.8 nao quer funcionar, testar uma versão mais antiga, enquanto nao é lançada uma mais recente, não fará mal.

**Comment by rafaelrandomize** [comment]
Very strange, maybe worth a try with GPG Suite 2022.2 for Mac OS by gpgtools.org, it uses an older version of gpg, 2.2.40, but if version 2.3.8 doesn't work, try a newer version old, while a newer one is not released, it won't do any harm.

**Comment by ArcaneSlime** [comment]
I am getting this error as well. Keys created both in gpg2 on Linux and Kleopatra on linux, both imported to openkeychain. Files encrypted with Kleopatra, with nautilus-seahorse, and by running gpg -u [user] --recipient [user] -se [file.zip] in the terminal get this issue, however by running gpg -u [user] --openpgp --recipient [user] -se [file.zip] I am able to successfully decrypt the file with openkeychain. I was doing some digging and found a seemingly difficult kleopatra dev (I think) that seemed to indicate this was because the PC clients use gpg 2.x.x whereas openkeychain uses an older version, and something about AEAD, but this is way out of my technical know-how, I don't even know what AEAD is. Hopefully at least knowing that passing the --openpgp flag works helps people in the meantime and can point to the direction of a real fix, but I sure would like to be able to encrypt from GUI.

**Comment by rafaelrandom** [comment]
> 

Hello friend, have you tried using gnupg-desktop-2.4.0.0-x86_64.AppImage? You can get it from the link https://download.gnupg.com/files/gnupg/gnupg-desktop-2.4.0.0-x86_64.AppImage, works great for me, I can encrypt and decrypt without problems between kleopatra and openkeychain.

**Comment by Fusted** [comment]
> > 
> 
> Hello friend, have you tried using gnupg-desktop-2.4.0.0-x86_64.AppImage? You can get it from the link https://download.gnupg.com/files/gnupg/gnupg-desktop-2.4.0.0-x86_64.AppImage, works great for me, I can encrypt and decrypt without problems between kleopatra and openkeychain.

I dont have problems with sync between linux and openkeychain, I cant sync between macOs and openkeychain

**Comment by rafaelrandom** [comment]
> > > 
> > 
> > 
> > Hello friend, have you tried using gnupg-desktop-2.4.0.0-x86_64.AppImage? You can get it from the link https://download.gnupg.com/files/gnupg/gnupg-desktop-2.4.0.0-x86_64.AppImage, works great for me, I can encrypt and decrypt without problems between kleopatra and openkeychain.
> 
> I dont have problems with sync between linux and openkeychain, I cant sync between macOs and openkeychain

Well, what is happening is that OpenKeychain is not able to decrypt files encrypted and signed via key pair by GPG versions from 2.3 onwards, only files encrypted with password by symmetric cryptography. Many Linux versions still use GPG version 2.2.xx as it is a Long Term Support version, and these versions are fully OpenKeychain compatible.

**Comment by rafaelrandom** [comment]
> > > 
> > 
> > 
> > Hello friend, have you tried using gnupg-desktop-2.4.0.0-x86_64.AppImage? You can get it from the link https://download.gnupg.com/files/gnupg/gnupg-desktop-2.4.0.0-x86_64.AppImage, works great for me, I can encrypt and decrypt without problems between kleopatra and openkeychain.
> 
> I dont have problems with sync between linux and openkeychain, I cant sync between macOs and openkeychain

I recommend you to use GPG Suite 2023.2 on MAC, it has GPG 2.2.41, OpenKeychain can perfectly decrypt files encrypted by this version of gpg.

**Comment by rafaelrandom** [comment]
@Fusted and @ArcaneSlime after some research and some testing, the most practical solution is to create the keypair with some version of gpg 2.2.xx. By importing these keys in gpg 2.3.xx or 2.4.xx, files encrypted in these versions of gpg can be decrypted in Openkeychain without problems. The other way, more uncomfortable, is if the keys were created with gpg 2.3.xx or 2.4.xx, you will only be able to decrypt the files in Openkeychain, if these keys are imported and the files are encrypted in gpg 2.2.xx.

**Comment by marcof333** [comment]
I get the same error in Android Samsung Tablet.
1. installed OpenKeyChain 5.8.2 in Samsung Tablet with Android 10
2. generated the key pair in OpenKeyChain
3. I try to decrypt a file encrypted and signed in a ubuntu pc but I get the same error:
"Processing input data
 Attempting to process OpenPGP data
 Encountered an error reading input data!"
 
 Any clue on how to solve this bug ? many thanks

**Comment by ArcaneSlime** [comment]
No, it seems this issue is considered "I don't care" by the dev, I can't even remember how long it has been since I posted that and no action seems to have been taken nor any indication that any ever will be. Basically the issue seems to be that the pgp version used by openkeychain is outdated, and everything else has updated. If you're on linux, you can do it through the terminal by passing the --openpgp flag while encrypting and it'll work, but you have to do it every time.

-------- Original Message --------
